src/App.tsx

```diff
@@ -1,14 +1,50 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterType } from './types/FilterType';
+
+export const titleCheck = (title: string, queryFilter: string) => {
+  return title.toLocaleLowerCase().includes(queryFilter.toLocaleLowerCase());
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [queryFilter, setQueryFilter] = useState('');
+  const [filterBy, setSelectStatus] = useState(FilterType.All);
+  const [todoId, setTodoId] = useState(0);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const todosFromServer = await getTodos();
+
+      setTodos(todosFromServer);
+      setIsLoading(false);
+    };
+
+    fetchData();
+  }, []);
+
+  const filteredTodos = todos.filter(({ title, completed }) => {
+    switch (filterBy) {
+      case FilterType.Active:
+        return !completed && titleCheck(title, queryFilter);
+
+      case FilterType.Completed:
+        return completed && titleCheck(title, queryFilter);
+
+      default:
+        return true;
+    }
+  });
```

Comment:
when filterBy = 'all' you don't check includes queryFilter in the title

Reply:
I have added the default check for 'All'

Reply:
Right, but on this default check you didn’t check query at all )